Core/Src/main.c

```diff
@@ -87,17 +89,46 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t RxData[3];
+  uint8_t TxData[3] = {0x01, 0x80, 0x00};
+  uint16_t ADC_Value;
+  float PWM_DutyCycle; //
+  uint8_t PWM_DutyCycle_Count;
+  const uint16_t PERIOD = 0xfa00; // 64000, our period, in hexadecimal
+
   /* USER CODE END 2 */
```

Comment:
You are missing a hal function call here that can ask the microcontroller to start the PWM wave as you configured in the ioc file. Unfortunately, if you don't call that function, no wave will be produced from that pin. I recommend searching about stm32 pwm and I will see the function I am talking about.